src/video/v4l_camera.cc

```diff
@@ -157,7 +157,7 @@ Video4LinuxCamera::open(const std::string &device,
 }
 
 void
-Video4LinuxCamera::enumerateControls(std::function<void(const v4l2_queryctrl &)> processControl)
+Video4LinuxCamera::enumerateControls(const std::function<void(const v4l2_queryctrl &)> &processControl)
```

Comment:
``std::function`` is pretty lightweight - I think it's ok to pass by value, does clang-tidy disagree?

Reply:
It does, though it may be being overzealous here.

Basically the options are:
1. disable the warning altogether (but I do like it, because it *did* find some places where we really should have been passing by reference)
2. add a magic comment to tell clang-tidy to ignore this particular case
3. turn it into a ``const`` reference

What do you think?

Reply:
I've just checked this, because I wasn't sure whether it made sense to take a ``const`` reference to a ``std::function`` in particular and it seems like you probably should: https://stackoverflow.com/questions/8711391/should-i-copy-an-stdfunction-or-can-i-always-take-a-reference-to-it?lq=1

It *could* be any arbitrary callable object, not just a function pointer.